src/backend/distributed/utils/node_metadata.c

```diff
@@ -620,27 +706,43 @@ AddNodeMetadata(char *nodeName, int32 nodePort, int32 groupId, char *nodeRack,
 		}
```

Comment:
True, I left the code how it was originally written. If I had to guess, the motivation was to make sure that not specifying always worked. By doing this we guarantee that `nextval(pg_dist_groupid_seq)` always returns an unused groupid.